### ISSUE TITLE
[P1][Sourcing][Sep18]Flickering when table has large row height in `virtualScroll`

### ISSUE THREAD

**Issue by CSENoni** [SAP/fundamental-ngx#10321]
#### Is this a bug, enhancement, or feature request?

bug

#### Briefly describe your proposal.

I have a platform table with `virtualScroll` on, and we use `fdp-textarea` in table cell. This issue is not related to specific component and I just have that component to set some certain height of table row. When scroll pass a default `renderAhead`, the table is little bit flickering. And when I hit a bottom of the table, the flickering can be easily observed:

https://github.com/SAP/fundamental-ngx/assets/22842279/72dccb13-cf6b-42ca-8307-c3aed25aea2a

It looks like `virtualScroll` can only work with certain `rowHeight` from api document. I tried to increase `rowHeight` but it did not help. 

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)

0.43.0

#### If this is a bug, please provide steps for reproducing it.

- Create a platform table with `virtualScroll` on.
- Have some inline components or some long text to increase row height.
- Scroll till it hits bottom

Please refer to the link below:

#### Please provide relevant source code if applicable.

https://stackblitz.com/edit/angular-td3x8a?file=src%2Fapp%2Fplatform-table-page-scrolling-example.component.html,package.json,src%2Fapp%2Fplatform-table-page-scrolling-example.component.ts

#### Is there anything else we should know?
In our application, we use inline component that may look like the `fdp-textarea` from the record. Also we can have multiple input fields in one cell that can increase row height.

**Comment by fkolar** [comment]
If you change this to:

```
 <fdp-textarea
        name="disabledDescription"
        [disabled]="true"
        [height]="'200px'"
        value="this is a disabled textarea"
      >
```

It does not even scroll smoothly at some point the scroll effect is replaced with replace effect

**Comment by droshev** [comment]
The mockup doesn't even look like a table. The advice is: make the table readonly while the user is scrolling. see the [comment](https://github.com/SAP/fundamental-ngx/issues/10242#issuecomment-1678624534)

**Comment by CSENoni** [comment]
@droshev That is the pagination table we are currently using in our application, and we are in the process of adopting NGX table to replace this one:
![Screenshot 2023-08-22 at 1 37 22 PM](https://github.com/SAP/fundamental-ngx/assets/22842279/f8b83a7d-9258-42a4-adfd-6547cea9e9ec)

Like I mentioned in the following comment, our user does not want that way and they are already familiar with having these components displaying and ready for them to use. Our UX team also mentioned that is not a good design for that table. In addition, through the fix in https://github.com/SAP/fundamental-ngx/issues/10242, it was actually a bug from the library since a single component was affecting a table. Ofc, we are aware of some trade-off here (even our pagination table has some flaws too). But if anything we could do to optimize or fix (clearly a bug) from the library, like `date-picker` from https://github.com/SAP/fundamental-ngx/issues/10242 , since that is not our component, please help us to address it. Thanks!

**Comment by CSENoni** [comment]
Sure, we will discuss further, but again this inline component scenario can be handled in UI5 sample: https://ui5.sap.com//#/entity/sap.ui.table.Table/sample/sap.ui.table.sample.Selection

**Comment by fkolar** [comment]
I believe there is something fundamental wrong with the Virtual Scroll. I agree it does not make sense to render 1000 controls as this is huge load on the browser, but still think that initial problem with the text area is a issue.

**Comment by droshev** [comment]
we need to downport it to `ng15`

**Comment by fkolar** [comment]
In terms of first case  - the text area. I get the same behavior even with Angular CDK virtual scroll (VS), even after setting different viewport buffers before and after. It seem VS has no issue with regular row heights but as soon as we try to include something higher that takes up more than 40% of the table height, than VS fails.  

https://github.com/fkolar/table-destroy

**Comment by CSENoni** [comment]
Even Angular team does not have a stable solution for this scenario: https://github.com/angular/components/issues/10113. It would be great if you can find a solution somehow, but with this scenario, our side may also need to find alternative way.

**Comment by N1XUS** [comment]
Platform virtual scroll calculates total rows height by multiplying `rowHeight` property by the amount of total rows. So in your case you can just pass custom number as `[rowHeight]` input property to define custom row height (with textarea). There was small issue with it, so please wait for latest release to fix it.

**Comment by droshev** [comment]
fixed with `0.43.6`

**Comment by CSENoni** [comment]
It works fine now without seeing any flickering if we know and set `rowHeight` beforehand. It does not work if row height is expanded by its content and larger than the fixed set. Also if we want the height is decided by each row content so that each row height is dynamically different, this will not be applicable. But that is different issue and also common issue from even angular cdk. For this issue, it works. Thanks!
